### ISSUE TITLE
Unable to update Intel graphics driver to 15.33.53.5161

### ISSUE THREAD

**Issue by rocketguedes** [jellysquid3/Sodium#2432]
### Bug Description

I cant start minecraft cause show this error telling me i have old driver for Intel, but I have a dedicated NVIDIA GTX 1050 ti, idk why sodium is using the integrated GPU, or caring about that. 

My CPU i5 3570 have a HD Graphics 2500, but im not using it. Has no HDMI connect to it. I just use the QSV on OBS.

![Captura de tela 2024-04-25 161739](https://github.com/CaffeineMC/sodium-fabric/assets/134816738/d959ce1b-a9e3-48b5-901e-dfc11003707a)

### Reproduction Steps

Just start from the Official Minecraft Launcher (Fabric Loader + Fabric API + Sodium + Iris)

### Log File

[latest.log](https://github.com/CaffeineMC/sodium-fabric/files/15119637/latest.log)

### Crash Report

no crash report, game not start

**Comment by pajicadvance** [comment]
We're already aware of this problem - a fix is currently in the works.

**Comment by jellysquid3** [comment]
The bug report's assumption as written is not correct. The problem is that Minecraft 1.20.5 changed the version of Oshi being used, and it now returns different version strings which Sodium doesn't know how to deal with. Not helping matters is that Intel uses both `10.18.10.5161` and `15.33.53.5161` in different places (why?!) to mean the same thing.

**Comment by rocketguedes** [comment]
> The bug report's assumption as written is not correct. The problem is that Minecraft 1.20.5 changed the version of Oshi being used, and it now returns different version strings which Sodium doesn't know how to deal with. Not helping matters is that Intel uses both `10.18.10.5161` and `15.33.53.5161` in different places (why?!) to mean the same thing.

You're right! Using [ATlauncher ](https://github.com/ATLauncher/ATLauncher) I created a 1.20.4 instance using the latest available mods versions and it has been started as expected with no crashes. The crash only happens in version 1.20.5.

**Comment by StupidProgrammer123** [comment]
please fix it it is becoming a issue

**Comment by StShAzzy** [comment]
![image](https://github.com/CaffeineMC/sodium-fabric/assets/90798128/c4d18e01-e695-48e2-ac56-a9f8676e1b13)
so any updates on this?

also before someone asks
![image](https://github.com/CaffeineMC/sodium-fabric/assets/90798128/c01856e9-af5a-4c20-832c-e12979a75d60)

**Comment by StShAzzy** [comment]
are you kidding me, that after taking a good hard look at the code, it already scans for GPUs yet *NOBODY* actually checking if there are any other GPUs installed on the system?
[this code already handles that](https://github.com/CaffeineMC/sodium-fabric/blob/d2a68153e5bb6ad59258db5b8e3604e8292f565b/src/main/java/net/caffeinemc/mods/sodium/client/compatibility/environment/probe/GraphicsAdapterProbe.java#L39C5-L39C12), yet no one actually tried doing something?
just add another check [here](https://github.com/CaffeineMC/sodium-fabric/blob/d2a68153e5bb6ad59258db5b8e3604e8292f565b/src/main/java/net/caffeinemc/mods/sodium/client/compatibility/checks/EarlyDriverScanner.java#L65) and you're done already

**Comment by Felix14-v2** [comment]
Create a PR if this is so simple...

**Comment by StShAzzy** [comment]
i will once i manage to test the changes, but that takes time from me, that i don't code for java, so i don't have anything to compile or test the code.
also this has been a issue for over 2 weeks now, it shouldn't take a stranger to come and fix it for them

**Comment by IMS212** [comment]
The problem is that at the point this check runs, we cannot know what GPU Minecraft is running on. It's not smart to assume it's running on the better GPU.

**Comment by StShAzzy** [comment]
it should be assumed, it's nonsensical to run a newer version of minecraft like 1.20 to hardware it doesn't support, most older iGPU from intel have buggy OpenGL support, so it's even another problem that should be avoided
but if you really insist on doing that, atleast add a config to ignore the result of the check so the game actually runs instead of not running at all unless you disable the iGPU or uninstall sodium and the mods that depend on it

**Comment by IMS212** [comment]
There is a config. You can pass in `-Dsodium.checks.win32.intelGen7=false`.

**Comment by StShAzzy** [comment]
well, i didn't actually knew, thank you

**Comment by jellysquid3** [comment]
You can be frustrated about not being able to use the mod due to a bug. But going on a tirade about how you supposedly know better, while having zero context as to why the code is written that way or the problems involved, is unacceptable.

The problem is *not* that we do not check for multiple GPUs. The problem is that OSHI (code that we do not write, and for which Minecraft updated) has changed the way they provide information about GPUs, and it no longer works reliably or returns the correct information. We have to replace a lot of code to fix these problems correctly.

**Comment by airpicnoobez** [comment]
> There is a config. You can pass in `-Dsodium.checks.win32.intelGen7=false`.

where do i type this in ?

**Comment by Poopooracoocoo** [comment]
@airpicnoobez If you're using Minecraft Launcher, go to the installations page, edit your Fabric installation, click 'more options', then add that argument to the end of your JVM arguments.

**Comment by hazardousrain** [comment]
are the drivers required or is there a way to bypass it? My laptop can not run the  drivers its asking for cuz im on a potato

**Comment by asumagic** [comment]
> are the drivers required or is there a way to bypass it? My laptop can not run the drivers its asking for cuz im on a potato

There is a way described above with the JVM flags but don't be surprised if it turns out completely broken.

Also, there is no real reason why "running on a potato" would prevent you from running those drivers if your iGPU is from the proper generation as described [in the wiki](https://github.com/CaffeineMC/sodium-fabric/wiki/Driver-Compatibility#windows-intel-gen7) (in the linked wikipedia table)... You'll have to seek support for this particular issue elsewhere.

**Comment by hazardousrain** [comment]
> > are the drivers required or is there a way to bypass it? My laptop can not run the drivers its asking for cuz im on a potato
> 
> There is a way described above with the JVM flags but don't be surprised if it turns out completely broken.
> 
> Also, there is no real reason why "running on a potato" would prevent you from running those drivers if your iGPU is from the proper generation as described [in the wiki](https://github.com/CaffeineMC/sodium-fabric/wiki/Driver-Compatibility#windows-intel-gen7) (in the linked wikipedia table)... You'll have to seek support for this particular issue elsewhere.
i have an i5 3210 but i can run 1.20.4 without problems. I cant use the latest drivers on that and the jvm arg didnt work
any other ideas please?

**Comment by Raidware** [comment]
> ### Bug Description
> I cant start minecraft cause show this error telling me i have old driver for Intel, but I have a dedicated NVIDIA GTX 1050 ti, idk why sodium is using the integrated GPU, or caring about that.
> 
> My CPU i5 3570 have a HD Graphics 2500, but im not using it. Has no HDMI connect to it. I just use the QSV on OBS.
> 
> ![Captura de tela 2024-04-25 161739](https://private-user-images.githubusercontent.com/134816738/325744726-d959ce1b-a9e3-48b5-901e-dfc11003707a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU1MTk5ODIsIm5iZiI6MTcxNTUxOTY4MiwicGF0aCI6Ii8xMzQ4MTY3MzgvMzI1NzQ0NzI2LWQ5NTljZTFiLWE5ZTMtNDhiNS05MDFlLWRmYzExMDAzNzA3YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxMlQxMzE0NDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wNDk0ZDFjMGE0MDczZDc1NjE4Yzk1NGY3NTk3MWYxOGZlNzRkNDMyOTZmMzQ4Y2QxZjY1YjVjNjNkYzAzODc3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.79kXtRLKJ3zZPNuAA8y9-Bw7whmWhpMzLLU6Fso9tqA)
> 
> ### Reproduction Steps
> Just start from the Official Minecraft Launcher (Fabric Loader + Fabric API + Sodium + Iris)
> 
> ### Log File
> [latest.log](https://github.com/CaffeineMC/sodium-fabric/files/15119637/latest.log)
> 
> ### Crash Report
> no crash report, game not start

same for me intel pentium 2117U processor

**Comment by jellysquid3** [comment]
We don't need additional "me too" posts. You have one of two options if you're **absolutely certain** your graphics drivers are up-to-date:

- Downgrade to Minecraft 1.20.4 (or 1.20.1) and wait for Sodium to publish an update which works around the bug in Minecraft.
- Add the JVM argument `-Dsodium.checks.win32.intelGen7=false` to your game's launch settings. This skips the check, but you will still run into problems if your drivers are not up-to-date.

**Comment by jellysquid3** [comment]
See issue #2498 for technical information/tracking on the problem.

**Comment by jellysquid3** [comment]
A lot of work has been done on fixing this. We've written [over 1000 lines of code](https://github.com/CaffeineMC/sodium-fabric/commit/a70d7b746cec2720e903f75bf9fb8dad17b48c9f) to replace the existing graphics adapter probe with something more robust. We need to carry out some additional testing but can release a hot fix in the coming days if everything works out.

**Comment by jellysquid3** [comment]
Fixed with https://github.com/CaffeineMC/sodium-fabric/compare/404151901893...b1164fb7c803. Will be backported and published here in the next few days.

**Comment by 0bsidianFox** [comment]
> There is a config. You can pass in `-Dsodium.checks.win32.intelGen7=false`.

Didn't work.

**Comment by BriayanRamdom** [comment]
bro, literally the intel hd 2500 support that actualization, and you have the benefit of more compatibility, just is download and install the intel hd 4000 drivers in his last version forced by the device administrator (does not broke or crash your pc)

**Comment by IAmSomeoneLol** [comment]
can somebody help me with the Sodium mod? I'm on 1.20.6 and I'm having the same issue to update the driver to a later version. Even though I'm using my Dedicated GPU instead of my IGPU.  The java arguments aren't working `-Dsodium.checks.win32.intelGen7=false`. it still pops up the exact same thing. 
**Sodium version:** Sodium 0.5.11

I tried forcing the game to go use my **Nvidia GPU** and not my **Intel HD4000** and it still doesn't work. Tho what strange is that I don't seem to have this issue on **Lunar Client** (_choose the sodium installation on it too_), so now idk what to do.

So how did **Lunar client fix this issue?????** I tried every version of **sodium for 1.20.6** (_all 3 of the releases_) and the exact same result! Only lunar client works!

Edit: Nevermind i checked Lunar Client's files, and turns out it uses sodium version 0.5.8, and for some reason my File Explorer was bugging and didn't delete the file when I was testing the other mods. So **Use sodium 0.5.8** as an easy fix till they update it and find the Corresponding mods that work for that specific version.
